## Part 1

**Null Hypothesis (Ho)**: 

No hay diferencia en la media de los Etching Rates para diferentes niveles de Power de los plasma beams.

**Alternate Hypothesis:** 

Hay diferencia significativa en la media de los Etching Rates para los diferentes niveles de Power.

**What is the significance level?** 

Se elige α=0.05. Esto implica un 5% de riesgo de rechazar la hipótesis nula cuando esta es cierta.

**What are the degrees of freedom of model, error terms, and total DoF?**

Dof_Model=2

DoF_Error=12

Dof_Total=14

## Part 2

In [1]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
import pandas as pd

In [2]:
data=pd.read_excel('data/anova_lab_data.xlsx', sheet_name='data_collected')
data

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [3]:
data.columns

Index(['Power ', 'Etching Rate'], dtype='object')

Renombramos las columnas para que funcione el modelo

In [4]:
data.rename(columns={'Power ':'Power', 'Etching Rate':'Etching_Rate'}, inplace=True)
data.columns

Index(['Power', 'Etching_Rate'], dtype='object')

In [5]:
model = ols('Etching_Rate ~ C(Power)',data=data).fit()
table = sm.stats.anova_lm(model)
print(table)

            df     sum_sq   mean_sq          F    PR(>F)
C(Power)   2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN


### Conclusions

Los resultados del ANOVA muestran un p-valor<α=0.05. Esto implica que se puede rechazar la Hipótesis Nula (Ho), es decir que
hay diferencia significativa en la media de los Etching Rates para los diferentes niveles de Power.

El F_estadístico que dio el modelo es ~ 36.88. Si vamos a la distribution table, utilizando el F_estadístico, el α y los grados de libertad, se llega a que el F_crítico es de 3.89 aproximadamente.

Por otra parte:

F_estadístico > F_crítico --> Nuevamente se rechaza Ho.

**Resumen:**
- α=0.05
- p-valor<α
- Grados de Libertad:
  - Dof_Model=2            
  - DoF_Error=12
  - Dof_Total=14
- F_estadístico=36.88
- F_crítico=3.89